creusot/src/translation/pearlite.rs

```diff
@@ -250,13 +250,30 @@ pub enum Literal<'tcx> {
 
 #[derive(Clone, Debug, TyDecodable, TyEncodable, TypeFoldable, TypeVisitable)]
 pub enum Pattern<'tcx> {
-    Constructor { variant: DefId, substs: GenericArgsRef<'tcx>, fields: Vec<Pattern<'tcx>> },
+    Constructor {
+        variant: DefId,
+        substs: GenericArgsRef<'tcx>,
+        fields: Vec<Pattern<'tcx>>,
+    },
+    /// Matches the pointed element of a pointer, so for `Box<T>` it matches `T`, for mutable borrows it matches the *current* value
+    Deref {
+        pointee: Box<Pattern<'tcx>>,
+        kind: PointerKind,
+    },
```

Comment:
This doesn't allow us to handle binding modes but it does give us potentially a little more flexibility in what we can do in source pearlite!